sw/Makefile

```diff
@@ -0,0 +1,74 @@
+SHELL = /usr/bin/env bash
+ROOT_DIR := /scratch/msc22f9/mempool/software
+SNITCH_DIR := /scratch/msc22f9/snitch
+APPS_DIR := $(ROOT_DIR)/apps
+HALIDE_DIR := $(ROOT_DIR)/halide
+RUNTIME_DIR := $(ROOT_DIR)/runtime
+TESTS_DIR := $(SNITCH_DIR)/sw/riscv-tests/isa
+TOOLCHAIN_DIR := $(abspath $(ROOT_DIR)/../toolchain)
+
+MEMPOOL_DIR       := $(shell git rev-parse --show-toplevel 2>/dev/null || echo $$MEMPOOL_DIR)
+INSTALL_DIR       ?= /scratch/msc22f9/mempool/install
+GCC_INSTALL_DIR   ?= $(INSTALL_DIR)/riscv-gcc
+SPIKE_INSTALL_DIR ?= $(INSTALL_DIR)/riscv-isa-sim
+
+RISCV_PREFIX ?= $(GCC_INSTALL_DIR)/bin/riscv32-unknown-elf-
+RISCV_GCC ?= $(RISCV_PREFIX)gcc
+RISCV_GCC_OPTS ?= -static -march=rv32imaXpulpimg -mcmodel=medany -fvisibility=hidden -nostdlib -nostartfiles
+# This will overwrite the ROOT_DIR variable from the included makefile
+#include $(RUNTIME_DIR)/runtime.mk
+include $(TESTS_DIR)/snitch_isa.mk
+
+# Applications
+.PHONY: apps
+apps:
+	make -C $(APPS_DIR) all
+
+.PHONY: halide-apps
+halide-apps:
+	make -C $(HALIDE_DIR) all
```

Comment:
Same as with the other makefiles; cleaner to centralize somewhere for tests than to edit this file.